status/condition_set.go

```diff
@@ -110,12 +110,14 @@ func (c ConditionSet) Set(condition Condition) (modified bool) {
 		} else {
 			// If we'd only update the LastTransitionTime, then return.
 			condition.LastTransitionTime = c.LastTransitionTime
+			condition.ObservedGeneration = c.ObservedGeneration
```

Comment:
So if the object generation changes here, I'm not sure that we can just bail, right? We have to always update to what the current object generation is here. So, if the object generation changes and we still don't change the reason/type/etc. we still need an update to happen here

Reply:
Synced offline, LastTransitionTime will not have to be updated along with observedGeneration. Updated.